tests/communication/test_decorator.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from concurrent.futures import ThreadPoolExecutor, as_completed
+import os
+import subprocess
+import sys
+import threading
+import unittest
+
+from maro.communication import Proxy, SessionMessage, dist
+from utils import get_random_port, proxy_generator
+
+
+def handler_function(that, proxy, message):
+    replied_payload = {"counter": message.payload["counter"] + 1}
+    proxy.reply(message, payload=replied_payload)
+    sys.exit(0)
+
+
+def lunch_receiver(proxy, handler_dict):
+    @dist(proxy, handler_dict)
+    class Receiver:
+        def __init__(self):
+            pass
+
+    receiver = Receiver()
+    receiver.launch()
+
+
+@unittest.skipUnless(os.environ.get("test_with_redis", False), "require redis")
+class TestDecorator(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        print(f"The dist decorator unit test start!")
+        # Prepare Redis.
+        redis_port = get_random_port()
+        cls.redis_process = subprocess.Popen(["redis-server", "--port", str(redis_port), "--daemonize yes"])
+        cls.redis_process.wait()
+
+        # Prepare proxy.
+        proxy_type_list = ["receiver", "sender"]
+        with ThreadPoolExecutor(max_workers=2) as executor:
```

Comment:
why we need a thread for receiver proxy?
we will lunch the receiver later.

Reply:
Those threads is working for the connection of proxies.

Reply:
but it's a little different from our real logic? Usually, the proxy and the local instance will be in the same thread.

Reply:
Yes. I will try to put the receive proxy and decorator in the same thread.